src/radical/pilot/agent/lm/spark.py

```diff
@@ -0,0 +1,285 @@
+
+__copyright__ = "Copyright 2016, http://radical.rutgers.edu"
+__license__   = "MIT"
+
+
+import os
+import subprocess
+import urllib
+import sys
+import socket
+import random
+
+from .base import LaunchMethod
+
+
+
+# ==============================================================================
+#
+# The Launch Method Implementation for Running Spark applications
+#
+class Spark(LaunchMethod):
+
+        # --------------------------------------------------------------------------
+    #
+    def __init__(self, cfg, logger):
+
+        LaunchMethod.__init__(self, cfg, logger)
+
+
+
+    # --------------------------------------------------------------------------
+    #
+    @classmethod
+    def lrms_config_hook(cls, name, cfg, lrms, logger):
+        """
+        FIXME: this config hook will inspect the LRMS nodelist and, if needed,
+               will start the SPARK cluster on node[0].
+        """
+        import radical.utils as ru
+
+        logger.info('Hook called by SPARK LRMS with the name %s'%lrms.name)
+
+        # If the LRMS used is not SPARK the namenode url is going to be
+        # the first node in the list and the port is the default one, else 
+        # it is the one that the SPARK LRMS returns
+        spark_home = None   
+        if lrms.name == 'SPARKLRMS':
+            logger.info("Found SPARK ")
+            logger.info('Hook called by SPARK LRMS')
+            logger.info('NameNode: {0}'.format(lrms.namenode_url))
+            rm_url         = "%s:%s" % (lrms.rm_ip, lrms.rm_port)
+            rm_ip          = lrms.rm_ip
+            launch_command = ru.which('spark')
+
+        else:
+            # Here are the necessary commands to start the cluster.
+            if lrms.node_list[0] == 'localhost':
+                node_name = lrms.node_list[0]
+            else:
+                logger.info("Bootstrap SPARK on " + socket.gethostname())
+                node_name =  socket.gethostname()
+            
+            VERSION = "1.5.2"
+            SPARK_DOWNLOAD_URL= "http://d3kbcqa49mib13.cloudfront.net/spark-1.5.2-bin-hadoop2.6.tgz" #prebuilt
+            #Download the tar file
+            opener = urllib.FancyURLopener({})
+            download_destination = os.path.join(os.getcwd(),"spark-" + VERSION + ".tar.gz")
+            logger.info("Download: %s to %s"%(SPARK_DOWNLOAD_URL, download_destination))
+            opener.retrieve(SPARK_DOWNLOAD_URL, download_destination)
+            spark_tar = "spark-" + VERSION + ".tar.gz"
+            if not os.path.isfile(spark_tar):
+                raise RuntimeError("Spark wasn't downloaded properly. Please try again")
+
+            os.system("tar -xzf" + spark_tar + "; rm " + spark_tar ) #untar and delete tarball 
+            os.system("mv spark-1.5.2-bin-hadoop2.6 spark-1.5.2")
+            spark_home = os.getcwd() + '/spark-' + VERSION
```

Comment:
The above lines mix strings which have `1.5.2` hard coded with strings which use `VERSION`.  You should decide for either option, otherwise `VERSION` is useless, really ;)